src/msg/simple/Pipe.cc

```diff
@@ -901,7 +901,8 @@ int Pipe::connect()
   // create socket?
   sd = ::socket(peer_addr.get_family(), SOCK_STREAM, 0);
   if (sd < 0) {
-    lderr(msgr->cct) << "connect couldn't created socket " << cpp_strerror(errno) << dendl;
+    rc = -errno;
+    lderr(msgr->cct) << "connect couldn't created socket " << cpp_strerror(rc) << dendl;
```

Comment:
I prefer to keep this change, since it is better to get errno saved before something might make another system call before cpp_strerror() is called now or in the future.